### PR TITLE
Tests for: LF-Support unicode syntax

### PR DESCRIPTION
**Pull request by vishwaksen-1**
Helping resolve issue #2551 

Adding necessary tests that complement with my latest PR in LF project where I added Unicode support. 

Please Check.

Thank you.

### CODE REVIEW

**Comment by vishwaksen-1** [comment]
I think the github actions are failing because this PR(new tests) strongly depends on the code addition in LF submodule.

**Review by facundominguez — COMMENTED** [review]
The parser tests look good, but I imagine that you also will want to add some test in `tests/basic/pos` to demonstrate that unicode operators can be used in LiquidHaskell annotations.

**Comment by vishwaksen-1** [comment]
Hi @facundominguez 
How does it look now? 
I didn't really understand what the tests in `tests/basic/pos` or `neg` do but I think they are like examples to demonstrate the new capability. Correct me if I'm wrong.

Thank you.

**Review by facundominguez — APPROVED** [review]
LGTM!

File: liquidhaskell-boot/tests/Parser.hs [hunk 1]
```diff
@@ -549,7 +569,7 @@ testErrorReporting =
               , "1 | ff :: {v:Nat  v >= 0 }"
               , "  |         ^"
               , "unexpected ':'"
```
Comment (facundominguez) [inline liquidhaskell-boot/tests/Parser.hs]: CI says
```
<test>:1:17:
  |
1 | ff :: {v:Nat  v >= 0 }
  |                 ^^
  | unexpected ">="
  | expecting "->", "<:", "=>", '|', '→', '⇒', bareTyArgP, mmonoPredicateP, or monoPredicateP
```
perhaps differences in library versions could change the error message. The test doesn't fail for me in my local development environment.

Comment (facundominguez) [reply liquidhaskell-boot/tests/Parser.hs]: I can reproduce the failure locally with
```
cabal update
cabal test --constraint="megaparsec>9.8.1" liquidhaskell-boot:liquidhaskell-parser
```
